### DISCUSSION TITLE
Passing output from Chromium route into LibreOffice route to compress images

### DISCUSSION THREAD

**Discussion by dafrica1** [gotenberg/gotenberg#961] [Q&A]
I'm implementing Gotenberg but see a pretty big tradeoff due to the limitations of Chromium and LibreOffice:

| Approach    | Pros | Cons |
| -------- | ------- | ------- |
| Chromium  | ✅ Respects HTML/CSS and fast at rendering    | ⛔ Resulting filesize can be huge depending on the images contained
| LibreOffice | ✅ Is able to compress images significantly resulting in size reductions of ~10x   | ⛔ Does not respect CSS and causes formatting to be different |

What I found was that if I created the PDF with the Chromium route first and then used the LibreOffice route to convert that PDF into another PDF (with some image compression settings), I could reduce the size of the PDF by almost 10x.

The problem is that it creates two files and all I care about is one. Not a huge deal but it would be great if there was a way to pipe the output of the Chromium route directly into the LibreOffice route.

Has anyone tried something like this before?

**Comment by gulien** [comment]
Hey @dafrica1 👋 

I was actually talking about this in https://github.com/gotenberg/gotenberg/issues/359#issuecomment-2310281734 👍 
I do agree, it would be better to pipe the resulting PDF of Chromium to LibreOffice. I'm already doing this for PDF/A for instance.

**Reply by dafrica1** [reply]
Thank you @gulien for all your hard work on Gotenberg! It's great to hear I'm not the only one.

If there was a form field in the Chromium route to enable  post-processing and then apply any user-defined LibreOffice specific options, that could be one way to handle it.

e.g.
```
curl \
--request POST http://localhost:3000/forms/chromium/convert/html \
--form files=@/path/to/index.html \
--form enablePostProcessing=true \  # This flag could enable the pipe to LibreOffice and submit the output to the LibreOffice route queue
--form quality=50 \
--form reduceImageResolution=true \
--form maxImageResolution=150 \
-o my.pdf
```
